### PR TITLE
[WIP] implement str.translate and str.maketrans

### PR DESCRIPTION
**Pull request by Furyzer0**
Currently `str.translate` is working. This is my first contribution to this project so I would like to know if there is something wrong in my code.
Ref: #190

### CODE REVIEW

File: vm/src/obj/objstr.rs [hunk 1]
```diff
@@ -1104,4 +1139,34 @@ mod tests {
             assert!(!PyString::from(s).istitle(&vm));
         }
     }
+
+    #[test]
+    fn str_translate() {
```
Comment (youknowone) [inline vm/src/obj/objstr.rs]: We also add python snippets under `tests/snippets`

**Comment by youknowone** [comment]
Instead of manual formatting, run `cargo fmt --all`. The broken test is that one.

File: vm/src/obj/objstr.rs [hunk 2]
```diff
@@ -829,6 +830,40 @@ impl PyString {
...
     }
+
+    // https://docs.python.org/3/library/stdtypes.html#str.translate
+    // Make it work for only subscribtable types
+    #[pymethod]
+    fn translate(&self, table: PyObjectRef, vm: &VirtualMachine) -> PyResult<String> {
+        let mut result = String::new();
+        for c in self.value.chars() {
+            let args = PyFuncArgs::new(vec![vm.new_int(c as i32)], vec![]);
+            match vm.call_method(&table, "__getitem__", args) {
```
Comment (adrian17) [inline vm/src/obj/objstr.rs]: `match table.get_item(c as i32, vm) {`

Comment (Furyzer0) [reply vm/src/obj/objstr.rs]: This looks way clearer!

**Comment by Furyzer0** [comment]
> Instead of manual formatting, run `cargo fmt --all`. The broken test is that one.

I was already using `cargo fmt`.

**Comment by coolreader18** [comment]
Are you using the latest rust toolchain?

File: vm/src/obj/objstr.rs (hunk 1 above)
Comment (Furyzer0) [reply vm/src/obj/objstr.rs]: > Are you using the latest rust toolchain?

You are right my default toolchain is stable which has different `rustfmt` than  `nightly`. Now formatting with `cargo +nightly fmt` and it seems all formatting problemsa are fixed now!

File: whats_left.sh [hunk 3]
```diff
@@ -8,4 +8,4 @@ python3 not_impl_gen.py
 
 cd ..
 
-cargo run -- tests/snippets/whats_left_to_implement.py
+cargo +nightly run -- tests/snippets/whats_left_to_implement.py
```
Comment (windelbouwman) [inline whats_left.sh]: Please leave this file as it is. Other people also depend upon this one.

**Comment by windelbouwman** [comment]
@Furyzer0 thank you for contributing to this project!

File: whats_left.sh (hunk 3 above)
Comment (Furyzer0) [reply whats_left.sh]: Ok I wasn't intented to push this change but I forgot that I changed it. I will revert it back with my next commit.

**Comment by Furyzer0** [comment]
It seems both functions work and all tests have passed. Tomorrow I will add the unit test for `str.maketrans` in the Rust module then it can be merged if there is no problem with the code.

**Pull request merged**